homeassistant/components/matrix/__init__.py

```diff
@@ -368,6 +370,28 @@ def _send_image(self, img, target_rooms):
                     ex.content,
                 )
 
+    def _send_video(self, vid, target_rooms):
+        _LOGGER.debug("Uploading file from path, %s", vid)
+
+        if not self.hass.config.is_allowed_path(vid):
+            _LOGGER.error("Path not allowed: %s", vid)
+            return
+        with open(vid, "rb") as upfile:
```

Comment:
I am wondering if there would be a benefit to using async for a big file read like this.